File: src/main/java/cabalgatas_salento/reservas/mapper/CaballoMapper.java
package cabalgatas_salento.reservas.mapper;

import cabalgatas_salento.reservas.dto.request.CaballoRequest;
import cabalgatas_salento.reservas.dto.response.CaballoResponse;
import cabalgatas_salento.reservas.entity.Caballo;
import org.springframework.stereotype.Component;

@Component
public class CaballoMapper {

    public CaballoResponse toResponse(Caballo caballo) {
        return CaballoResponse.builder()
                .id(caballo.getId())
                .nombre(caballo.getNombre())
                .raza(caballo.getRaza())
                .isActive(caballo.getIsActive())
                .eliminado(caballo.getEliminado())
                .build();
    }

    public Caballo toEntity(CaballoRequest request) {
        return Caballo.builder()
                .nombre(request.getNombre())
                .raza(request.getRaza())
                .build();
    }
}

File: src/caballos/mappers/caballo.mapper.ts
import { Caballo } from '../entities/caballo.entity';
import { CaballoResponseDto } from '../dto/caballo-response.dto';

export class CaballoMapper {
  static toResponseDto(entity: Caballo): CaballoResponseDto {
    const dto = new CaballoResponseDto();
    dto.id = entity.id;
    dto.nombre = entity.nombre;
    dto.raza = entity.raza;
    dto.isActive = entity.isActive;
    dto.eliminado = entity.eliminado;
    return dto;
  }
}
